p2p/libp2p/networksharding/sharder.go

```diff
@@ -40,15 +39,15 @@ type Sharder interface {
 	// GetDistance get the distance between a and b
 	GetDistance(a, b sortingID) *big.Int
```

Comment:
Trying to reuse the Sharder interface & implementations elsewhere and found out that we might change the interface and implementations in a following manner:
```GO
type Sharder interface {
     CanConnectToPeer(newPeer peer.ID, currentList []peer.ID) bool
     CanClosePeer(newPeer peer.ID, currentList []peer.ID) bool
     CanCallBootstrap() bool
}
```

In this way, the clients of this interface (KadDhtDiscoverer, libp2pConnectionMonitor and possible KadDhtHost) will only need this component to tell them what to do: resume the discovery process, closing unwanted peers and if it's worth the connection to be really made. The logic with all the required parameters, watchdog, kademlia implementation, sorting, bit masking and so on will be encapsulated in the kad sharder implementation(s). 
Sorting is just too general as we need the target peers to be inserted just about everywhere ending up having different portions of duplicated code.

Reply:
OK, but in a future PR

Reply:
Ok, will be done in a future PR.